### ISSUE TITLE
Linux: configure doesn't accept go version 1.10

### ISSUE THREAD

**Issue by woguls** [swig/swig#1206]
I have installed the currently "featured version" of go version 1.10, via https://golang.org/dl/
When generating the configure script I get:

```
checking for go... go
checking whether go version is too old... yes - minimum version is 1.2
```

For the moment I am able to manually change the configure script to accept go1.10 and everything seems to work.

**Comment by ojwb** [comment]
Looks like the pattern in `configure.ac` doesn't expect two digit minor versions:

```
   case $go_version in
    go1.1* | go1.0* | go1 )
      AC_MSG_RESULT([yes - minimum version is 1.2])
```

That needs fixing, but should only mean that you can't run SWIG's testsuite for Go - you should still get a SWIG build that can generate Go bindings.

**Comment by ojwb** [comment]
Fixed by e57d845312131320dc5be91f97f693f0a10c548e.
